contrib/inventory/vultr.py

```diff
@@ -0,0 +1,180 @@
+ - vultr:
+   A group containing all hosts
+ - tag:
+   A group is created for each tag. E.g. groups 'tag_foo' and 'tag_bar' are created if there exist
+   instances with tags 'foo' or 'bar'.
+ - location:
+   A group is created for each location. E.g. group 'location_Chicago' is created if a server resides
+   in Chicago.
+
+Examples:
+  Execute uname on all instances in location 'Chicago'
+  $ ansible -i vultr.py location_Chicago -m shell -a "/bin/uname -a"
+
+  Install nginx on all Red Hat web servers tagged with 'www'
+  $ ansible -i vultr.py tag_www -m yum -a "name=nginx state=present"
+
+  Run site.yml playbook on web servers
+  $ ansible-playbook -i vultr.py site.yml -l tag_www
+
+Support:
+  This script is tested on Python 2.7 and 3.4. It may work on other versions though.
+
+Author: Keth Resar <Keith.Resar@422long.com>
+Version: 0.1
+"""
+
+
+import sys
+import os
+
+class VultrApiException(Exception):
+    pass
+
+try:
+    from ConfigParser import SafeConfigParser as ConfigParser
+except ImportError:
+    from configparser import ConfigParser
+
+try:
+    import json
+except ImportError:
+    import simplejson as json
+
+try:
+    import requests
+except:
+    sys.exit('Vultr.com inventory script requires requests. See http://docs.python-requests.org/en/master/')
+
+
+class VultrInventory:
+
+    _API_ENDPOINT = 'https://api.vultr.com'
+
+    def __init__(self):
+        self._configure_from_file()
+        self.inventory = self.get_inventory()
+
+    def _configure_from_file(self):
+        """Initialize from .ini file.
+
+        Configuration file is assumed to be named 'vultr.ini' and to be located on the same
+        directory than this file, unless the environment variable BROOK_INI_PATH says otherwise.
+        """
+
+        try:
+            vultr_ini_default_path = \
+                os.path.join(os.path.dirname(os.path.realpath(__file__)), 'vultr.ini')
+            vultr_ini_path = os.environ.get('VULTR_INI_PATH', vultr_ini_default_path)
+    
+            config = ConfigParser(defaults={
+                'api_token': '',
+            })
+            config.read(vultr_ini_path)
+            self.api_token = config.get('vultr', 'api_token')
+        except:
+            pass
+
+        if 'VULTR_API_TOKEN' in os.environ:  self.api_token = os.environ.get("VULTR_API_TOKEN")
+
+        if not self.api_token:
+            print('You must provide (at least) your Vultr.com API token to generate the dynamic '
+                  'inventory.')
+            sys.exit(1)
+
+
+    def get_inventory(self):
+        """Generate Ansible inventory.
+        """
+
+        groups = { 'vultr': [], '_meta': { 'hostvars': { } } }
+
+        r = requests.get('%s/v1/server/list' % self._API_ENDPOINT, headers={ 'API-Key': self.api_token })
+
+        if r.status_code != 200:  raise(VultrApiException("Error listing servers"))
+
+
+        # Build inventory from instances in all projects
+        #
+        for server in r.json().values():
```

Comment:
This fails with `AttributeError: 'list' object has no attribute 'values'` if there are currently no servers deployed at vultr and the API query returns an empty json document `[ ]`